dpctl/tensor/libtensor/source/tensor_py.cpp

```diff
@@ -1879,6 +1879,290 @@ eye(py::ssize_t k,
                           eye_event);
 }
 
+/* =========================== Tril and triu ============================== */
+// define function type
+typedef sycl::event (*tri_fn_ptr_t)(sycl::queue,
+                                    py::ssize_t,   // inner_range  //py::ssize_t
+                                    py::ssize_t,   // outer_range
+                                    char *,        // src_data_ptr
+                                    char *,        // dst_data_ptr
+                                    py::ssize_t,   // nd
+                                    py::ssize_t *, // shape_and_strides
+                                    int,           // k
+                                    const std::vector<sycl::event> &,
+                                    const std::vector<sycl::event> &);
+
+template <typename Ty, bool> class tri_kernel;
+template <typename Ty, bool l>
+sycl::event tri_impl(sycl::queue exec_q,
+                     py::ssize_t inner_range,
+                     py::ssize_t outer_range,
+                     char *src_p,
+                     char *dst_p,
+                     py::ssize_t nd,
+                     py::ssize_t *shape_and_strides,
+                     int k,
+                     const std::vector<sycl::event> &depends,
+                     const std::vector<sycl::event> &additional_depends)
+{
+    constexpr int d2 = 2;
+    py::ssize_t src_s = nd;
+    py::ssize_t dst_s = 2 * nd;
+    py::ssize_t nd_1 = nd - 1;
+    py::ssize_t nd_2 = nd - 2;
+    Ty *src = reinterpret_cast<Ty *>(src_p);
+    Ty *dst = reinterpret_cast<Ty *>(dst_p);
+
+    sycl::event tri_ev = exec_q.submit([&](sycl::handler &cgh) {
+        cgh.depends_on(depends);
+        cgh.depends_on(additional_depends);
+        cgh.parallel_for<tri_kernel<Ty, l>>(
+            sycl::range<2>(inner_range, outer_range), [=](sycl::item<2> idx) {
```

Comment:
Please rework this to use `sycl::range<1>(inner_range * outer_range)`. Please use `sycl::id<1>` rather than `sycl::item<1>` since the code does not use additional information exposed by `item<1>`. This will reduce amount of data copied to the kernel during submission. 

Assuming 1D range, and use of `sycl::id<1>` one can recover `inner_gid` and `outer_gid` as follows:

```
    py::ssize_t outer_gid = idx[0] / inner_range;
    py::ssize_t inner_gid = idx[0] - inner_range * outer_gid;
```

This change resulted, in my local experiments, the `%time Y = dpt.tril(X)` for `X = X = dpt.ones((15013, 50, 50), 'i4')` to drop about 30% Iris Xe-LP.